blockchain/chain_test.go

```diff
@@ -110,3 +111,371 @@ func TestHaveBlock(t *testing.T) {
 		}
 	}
 }
+
+// TestCalcSequenceLock tests the LockTimeToSequence function, and the
+// CalcSequenceLock method of a Chain instance. The tests exercise several
+// combinations of inputs to the CalcSequenceLock function in order to ensure
+// the returned SequenceLocks are correct for each test instance.
+func TestCalcSequenceLock(t *testing.T) {
+	// Lock up the first 4 blocks of the main chain.
```

Comment:
A test chain is needed for the tests since `SequenceLocks` hits the DB to fetch the headers if the relative lock is seconds-based.